security/pkg/server/ca/server.go

```diff
@@ -39,8 +39,8 @@ type CertificateAuthority interface {
 	// Sign generates a certificate for a workload or CA, from the given CSR and cert opts.
 	Sign(csrPEM []byte, opts ca.CertOpts) ([]byte, error)
 	// SignWithCertChain is similar to Sign but returns the leaf cert and the entire cert chain.
-	SignWithCertChain(csrPEM []byte, opts ca.CertOpts) ([]byte, error)
-	// GetCAKeyCertBundle returns the KeyCertBundle used by CA.
+	SignWithCertChain(csrPEM []byte, opts ca.CertOpts) ([]string, error)
```

Comment:
Also seems we should not have `Sign` if its never used...

Reply:
this is mainly because the grpc security service responds with three strings, the root-cert, the cert-chain and the main cert. its difficult to contruct this from just the byte array. IF this is purely used for the RA code, I think it should be ok to change the signature?

Reply:
Oh I didn't realize it was a `[]string`. Misread as `string`